packages/core/src/app/client/wallet-db.ts

```diff
@@ -0,0 +1,70 @@
+/* Internal Imports */
+import { WalletDB, Keystore, KeyValueStore } from '../../interfaces'
+
+export class DefaultWalletDB implements WalletDB {
+  /**
+   * Initializes the database wrapper.
+   * @param db Database instance to insert data into.
+   */
+  constructor(private db: KeyValueStore) {}
+
+  /**
+   * Adds a keystore file to the database.
+   * @param keystore Keystore file to add.
+   * @returns a Promise that resolves once the keystore has been inserted.
+   */
+  public async putKeystore(keystore: Keystore): Promise<void> {
+    const key = Buffer.from(keystore.address)
+    const value = Buffer.from(JSON.stringify(keystore))
+
+    if (await this.hasKeystore(keystore.address)) {
```

Comment:
Don't know the context in which it's used, but I think there may be a race condition here.

Reply:
I think it should be fine this instance since there's (effectively) a 1:1 mapping between addresses and keystore files.

Reply:
Honestly could probably just remove the error check, no real reason to have it there.